store/sdk/src/main/java/org/apache/carbondata/sdk/file/Field.java

```diff
@@ -55,7 +55,7 @@
    * @param type datatype of field, specified in strings.
    */
   public Field(String name, String type) {
-    this.name = name;
+    this.name = name.toLowerCase().trim();
```

Comment:
CarbonWriterBuilder.updateSchemaFields() is already converting to lowercase, just add trim in that method. No need to handle for each here.